### DISCUSSION TITLE
Automatic split threshold behavior

### DISCUSSION THREAD

**Discussion by aolko** [FancyWM/fancywm#254] [Ideas]
New automatic threshold  is cool and all, but let users choose the behavior: vertical|horizontal|stack|none

**Comment by veselink1** [comment]
Thanks for the suggestion. Currently, a split in the opposite direction of the parent is created when the threshold is reached, but that could be made more configurable as you suggest.

From the options you've mentioned, I guess `none` is already covered by increasing the auto-split threshold? 

I'm not sure selecting between vertical/horizontal makes sense - I suspect you'd want to automatically determine V/H split based on the parent. If the split is always oriented the same, you just end up with a row of windows in nested panels.

So based on the above, there could be an option to toggle between either creating stack panel or split, but I'm not sure a fixed "vertical/horizontal" option makes sense to me. Perhaps if you could give an example?

**Reply by aolko** [reply]
> From the options you've mentioned, I guess `none` is already covered by increasing the auto-split threshold?

You should let the user to turn it off

> I'm not sure selecting between vertical/horizontal makes sense - I suspect you'd want to automatically determine V/H split based on the parent. If the split is always oriented the same, you just end up with a row of windows in nested panels.
> 
> So based on the above, there could be an option to toggle between either creating stack panel or split, but I'm not sure a fixed "vertical/horizontal" option makes sense to me. Perhaps if you could give an example?

Okay, how about this: default|vertical|horizontal|stack|none?
Vertical and horizontal **only** split down one axis, either x or y.

**Reply by javesar** [reply]
Veselink1 I understand what you mean by
> I'm not sure selecting between vertical/horizontal makes sense
For me it does make sense because it'd be nice to group windows that are related to the same task in a panel, even if it's in the same direction of the parent. Also, I have a 32:9 monitor so I naturally default to vertical stacks
